file/validate.go

```diff
@@ -29,5 +82,16 @@ func validate(content []byte) error {
 		err := fmt.Errorf(desc.String())
 		errs.Errors = append(errs.Errors, err)
 	}
+
+	var kongdefaults *KongDefaults
+	if err := json.Unmarshal(content, kongdefaults); err != nil {
+		return fmt.Errorf("unmarshaling file into KongDefaults: %w", err)
+	}
+	if kongdefaults != nil {
+		if res := checkDefaults(*kongdefaults); !res {
+			return fmt.Errorf("fields are not allowed to specify in defaults")
+		}
+	}
```

Comment:
Once the `checkDefaults` function returns []error, please merge errors returned from jsonSchema library with this result and return the whole thing.